baseapp/baseapp.go

```diff
@@ -493,15 +493,15 @@ func validateBasicTxMsgs(msgs []sdk.Msg) sdk.Error {
 	return nil
 }
 
+// retrieve the context for the ante handler and store the tx bytes; store
+// the signing validators if the tx runs within the deliverTx() state.
 func (app *BaseApp) getContextForAnte(mode runTxMode, txBytes []byte) (ctx sdk.Context) {
 	// Get the context
-	if mode == runTxModeCheck || mode == runTxModeSimulate {
-		ctx = app.checkState.ctx.WithTxBytes(txBytes)
-	} else {
-		ctx = app.deliverState.ctx.WithTxBytes(txBytes)
-		ctx = ctx.WithSigningValidators(app.signedValidators)
+	ctx = getState(app, mode).ctx.WithTxBytes(txBytes)
+	if mode != runTxModeDeliver {
```

Comment:
(nit) Writing this with `if mode == runTxModeDelvier` is a bit more obvious (that we're doing something special for delivertx)

Reply:
Incorporated, thanks